codegen/module_system.go

```diff
@@ -544,6 +573,57 @@ func (g *httpClientGenerator) Generate(
 	}, nil
 }
 
+// PopulateQPSLevels loops through endpoint dir and gets qps levels
+func PopulateQPSLevels(EndpointsBaseDir string) map[string]int {
+	qpsLevels := make(map[string]int)
+	endpointFiles := GetListOfAllFilesInEndpointDir(EndpointsBaseDir)
+	for _, endpointFile := range endpointFiles {
+		var config map[string]interface{}
+		bytes, _ := ioutil.ReadFile(endpointFile)
+		err := yaml.Unmarshal(bytes, &config)
+		if err != nil {
+			// try json
+			_ = json.Unmarshal(bytes, &config)
+		}
+		clientMethod, methodOK := config["clientMethod"]
+		clientID, clientOK := config["clientId"]
+		qpsLevel, qpsOK := config["qpsLevel"]
+		if methodOK && clientOK && qpsOK && err == nil {
+			// edge case where clientID or clientMethod is nil
+			if clientID != nil && clientMethod != nil && clientID != "" && clientMethod != "" {
+				// unique key because of potential clients having same method names (staging)
+				key := clientID.(string) + "-" + clientMethod.(string)
+				// store highest qps level for circuit breaker in qpsLevels map
+				thisQPSLevel := int(qpsLevel.(float64))
```

Comment:
This should be int instead of float?

Reply:
It parses it as a float so I had to convert it to an int